source/docs/software/vscode-overview/3rd-party-libraries.rst

```diff
@@ -39,49 +39,56 @@ How Does It Work? - Python
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      **Checking for Updates (Offline)**
 
-Part of the JSON file that vendors may optionally populate is an online update location. If a library has an appropriate location specified, running :guilabel:`Check for updates (online)` will check if a newer version of the library is available from the remote location.
+      Since dependencies are version managed on a per-project basis, even when installed offline, you will need to :guilabel:`Manage Vendor Libraries` and select :guilabel:`Check for updates (offline)` for each project you wish to update.
 
-Removing a Library Dependency
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      **Checking for Updates (Online)**
 
-To remove a library dependency from a project, select :guilabel:`Manage Current Libraries` from the :guilabel:`Manage Vendor Libraries` menu, check the box for any libraries to uninstall and click :guilabel:`OK`. These libraries will be removed as dependencies from the project.
+      Part of the JSON file that vendors may optionally populate is an online update location. If a library has an appropriate location specified, running :guilabel:`Check for updates (online)` will check if a newer version of the library is available from the remote location.
 
-Command-Line
-~~~~~~~~~~~~
+      **Removing a Library Dependency**
+
+      To remove a library dependency from a project, select :guilabel:`Manage Current Libraries` from the :guilabel:`Manage Vendor Libraries` menu, check the box for any libraries to uninstall and click :guilabel:`OK`. These libraries will be removed as dependencies from the project.
+
+      **Command-Line**
+
+      Adding a vendor library dependency from the vendor URL can also be done through the command-line via a gradle task. Open a command-line instance at the project root, and enter ``gradlew vendordep --url=<url>`` where ``<url>`` is the vendor JSON URL. This will add the vendor library dependency JSON file to the ``vendordeps`` folder of the project. Vendor libraries can be updated the same way.
+
+      The ``vendordep`` gradle task can also fetch vendordep JSONs from the user ``wpilib`` folder. To do so, pass ``FRCLOCAL/Filename.json`` as the file URL. For example, ``gradlew vendordep --url=FRCLOCAL/WPILibNewCommands.json`` will fetch the JSON for the command-based framework.
+
+   .. tab-item:: Python
+      :sync: python
+
+      All RobotPy project dependencies are specified in ``pyproject.toml``. You can add additional vendor-specific dependencies either by:
 
-Adding a vendor library dependency from the vendor URL can also be done through the command-line via a gradle task. Open a command-line instance at the project root, and enter ``gradlew vendordep --url=<url>`` where ``<url>`` is the vendor JSON URL. This will add the vendor library dependency JSON file to the ``vendordeps`` folder of the project. Vendor libraries can be updated the same way.
+      * Adding the component name to ``robotpy_extras``
```

Comment:
This lists these two ways, but then down below it always uses the robotpy_extras or running a pip command, which doesn't seem to mentioned here, unless the pip command adds it to requires (dumb question from not a python guy)

Reply:
I see what you mean, changed it up a bit, tell me if that's clearer.